### DISCUSSION TITLE
How to use React hook form with react flag select

### DISCUSSION THREAD

**Discussion by mernaff18** [react-hook-form/react-hook-form#8007] [Q&A] [answered]
I tried to implement React Flags Select with React Hook From using Controller . While I am submitting the form country filed return undefined. 

```
<Controller name="country"
    			control={control}
    			render={({ field: { onChange, value } }) => (
    				<ReactFlagsSelect
    					selected={selected}
    					onSelect={code => handleChange(code)}
    					value={value}
    					onChange={onChange}
				/>
			)}
/>
```

**Comment by bluebill1049** [comment]
could you provide a codesandbox?

**Reply by mernaff18** [reply]
https://codesandbox.io/s/dreamy-forest-1y6m23

above is my link.. while submitting form country field value is undefined

**Reply by mernaff18** [reply]
Now i am able fetch value.. now issue is controller fields are not resetting. 

https://codesandbox.io/s/dreamy-forest-1y6m23

Thanks,
Royce.L

**Comment by Moshyfawn** [comment, marked as answer]
From [the RFS GH page docs](https://github.com/ekwonye-richard/react-flags-select), there're no `value` and `onChange` props on the `ReactFlagsSelect` component. This where you might be overriding your RHF values by providing undefined instead.

See [this RHF minimal usage example with RFS integration](https://codesandbox.io/s/modern-browser-udlqkx?file=/src/App.tsx)
